### ISSUE TITLE
ObjectStore.js fail to handle null in Number datatype

### ISSUE THREAD

**Issue by kidwm** [ParsePlatform/ParseReact#61]
```js
var first = new Parse.Object('Item');
first.set('quantity', 0); // first set Datatype to Number
first.save();

var second = new Parse.Object('Item');
second.set('quantity', NaN); // `NaN` would be store as `null` in Database
second.save();

observe: function(props) { // this `null` would trigger error in ObjectStore.js
	return {
		items: (new Parse.Query('Item')).greaterThan('quantity', 0)
	};
}
```

**Comment by mikemintz** [comment]
This might be fixed by https://github.com/ParsePlatform/ParseReact/pull/58 which was merged in last week. Have you tried with the latest version of ParseReact 0.3.1?

Or is the problem that you don't want NaN to be coerced to null when you save the object?

**Comment by kidwm** [comment]
@mikemintz Yes, I'm using 0.3.1

I don't think the coercion done by parse-js-sdk is wrong.
I just want ObjectStore.js could handle this situation and does not make query failed and break the app.

**Comment by andrewimm** [comment]
`NaN` is converted to `null` in our database, because JSON does not support IEEE 754 concepts like `NaN` or `Infinity`.

I will attempt to reproduce this bug, but it would be easier if you could give me the full stack trace from ObjectStore.js

**Comment by kidwm** [comment]
@mikemintz You're right, this issue has been fixed in 0.3.1, I didn't noticed that I had not upgraded it in my production environment.

@andrewimm Thanks!
